dali/util/fits.cc

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/util/fits.h"
+#include <fitsio.h>
+#include <string>
+#include <vector>
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+namespace fits {
+
+void HandleFitsError(int status) {
+  std::string status_str;
+  status_str.reserve(FLEN_STATUS);
```

Comment:
I think formally it should be `resize`, as `reserve` does not change the `size` of the string (see https://en.cppreference.com/w/cpp/string/basic_string/operator_at does not guarantee anything with respect to capacity).

Reply:
Also you could do the resize inside the `if`.